docs/setup/install-existing.md

```diff
@@ -107,22 +149,161 @@ The [aad-pod-identity](https://github.com/Azure/aad-pod-identity) gives a clean
+Refer to the [tutorials](../tutorial.md) to understand how you can expose an AKS service over HTTP or HTTPS, to the internet, using an Azure App Gateway.
+
+## Hybrid Environment
+AGIC will assume full ownership of the AGIC configuration. What if we'd like to manually add a listener and a backend?
+We have the option to constrain AGIC, so it controls only a subset of the App Gateway's properties.
+
+### Scenario
+Let's take an imaginary App Gatway, which manages traffic for 2 web sites:
+  - prod.contoso.com
+  - dev.contoso.com
+
+We need to:
+  - start serving `dev.contoso.com` from a new AKS, using App Gateway and AGIC
+  - continue serving `prod.contoso.com` from an existing [Azure VMSS](https://azure.microsoft.com/en-us/services/virtual-machine-scale-sets/)
+
+Until now (and by default) AGIC assumed 100% ownership of the App Gateway configuration. AGIC overwrites all of App
+Gateway's configuration. If we were to manually create a listener for `prod.contoso.com` (on App Gateway), without
+defining it in the Kubernetes Ingress, AGIC will delete the `prod.contoso.com` config within seconds.
+
+To install AGIC and also serve `prod.contoso.com` from our VMSS machines, we must create the following
+[CRD](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/):
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: "appgw.ingress.k8s.io/v1"
+kind: AzureIngressProhibitedTarget
+metadata:
+  name: prod-contoso-com
+spec:
+  hostname: prod.contoso.com
+EOF
+```
+
+The bash command above will create an `AzureIngressProhibitedTarget` object. This makes AGIC aware of the existence of
+App Gateway config for `prod.contoso.com`. It explicitly instructs App Gateway to avoid changing any configuration
+related to the `hostname` in the command above.
+
+
+### Enable brownfield deployment for a new AGIC installation
+To limit AGIC to a subset of the configuration of your App Gateway, we need to modify the `helm-config.yaml` template.
+Set `brownfield.enabled` to "true" in `helm-config.yaml`:
+
+```yaml
+brownfield:
+    enabled: false
+```
+
+This instructs Helm to:
+  - create a new CRD: `AzureIngressProhibitedTarget`
+  - create new instance of `AzureIngressProhibitedTarget` called `prohibit-all-targets`
+
+The default `prohibit-all-targets` prohibits AGIC from changing config for *any* host and path. Helm install
+with `brownfield.enabled=true` will deploy AGIC, but will not make any changes to App Gateway.
+
+
+### Broaden permissions
+Enabling `brownfield.enabled=true` and `helm` installing AGIC, would result in the creation of a `prohibit-all-targets` object in your AKS.
+
+View the contents of the object:
+```bash
+kubectl get AzureIngressProhibitedTarget prohibit-all-targets -o yaml
+```
+
+You can broaden the permissions AGIC has:
+1. Create a new `AzureIngressProhibitedTarget` with your specific setup:
+    ```bash
```

Comment:
Shouldn't this be enough ?
```bash
helm upgrade \
          --recreate-pods \
          --version 0.8.0 \
          -f .dev/helm-config.yaml \
          --set appgw.shared=true
          ingress-azure application-gateway-kubernetes-ingress/ingress-azure
```

Reply:
It turns out - not.  I ran into this:
```
[debug] Created tunnel using local port: '61149'

[debug] SERVER: "127.0.0.1:61149"

UPGRADE FAILED
ROLLING BACK
Error: failed to create resource: the server could not find the requested resource (post azureingressprohibitedtargets.appgw.ingress.k8s.io)
Error: UPGRADE FAILED: failed to create resource: the server could not find the requested resource (post azureingressprohibitedtargets.appgw.ingress.k8s.io)
```
Perhaps this is a matter of properly configuring the helm upgrade scenario.

Reply:
And after I create the CRD - `helm` then creates `prohibit-all-targets` as expected.